device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -1560,7 +1560,7 @@ private void updateStatus(IotHubConnectionStatus newConnectionStatus, IotHubConn
             log.debug("Invoking connection status callbacks with new status details");
             invokeConnectionStateCallback(newConnectionStatus, reason);
 
-            if (deviceClientConfigs.size() < 2 || newConnectionStatus != IotHubConnectionStatus.CONNECTED)
+            if (!isMultiplexing || newConnectionStatus != IotHubConnectionStatus.CONNECTED)
             {
```

Comment:
is "newConnectionStatus" the state of the mux client => tcp connection?

Reply:
Yes. Within the context of this method, the ```newConnectionStatus``` argument is the TCP level connection